src/script/descriptor.cpp

```diff
@@ -514,7 +514,7 @@ class AddressDescriptor final : public DescriptorImpl
     std::vector<CScript> MakeScripts(const std::vector<CPubKey>&, const CScript*, FlatSigningProvider&) const override { return Vector(GetScriptForDestination(m_destination)); }
 public:
     AddressDescriptor(CTxDestination destination) : DescriptorImpl({}, {}, "addr"), m_destination(std::move(destination)) {}
-    bool IsSolvable() const final { return false; }
+    bool IsSolvable() const override final { return false; }
```

Comment:
This is final, but not virtual, so override is already implied. I am not sure why the compiler warns here?!

Reply:
GCC (Debian 8.3.0-6) will warn about this:
```bash
  CXX      script/libbitcoin_common_a-descriptor.o
script/descriptor.cpp:517:10: warning: 'virtual bool {anonymous}::AddressDescriptor::IsSolvable() const' can be marked override [-Wsuggest-override]
     bool IsSolvable() const final { return false; }
          ^~~~~~~~~~
```

However `clang-tidy` with [modernize-use-override](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-override.html) will tell you that it, along with a bunch of the other overrides added here are redundant:
```bash
src/script/descriptor.cpp:378:10: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
    bool IsRange() const override final
         ^               ~~~~~~~~~
src/script/descriptor.cpp:414:17: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
    std::string ToString() const override final
                ^                ~~~~~~~~~
src/script/descriptor.cpp:421:10: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
    bool ToPrivateString(const SigningProvider& arg, std::string& out) const override final
         ^                                                                   ~~~~~~~~~
src/script/descriptor.cpp:480:10: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
    bool Expand(int pos, const SigningProvider& provider, std::vector<CScript>& output_scripts, FlatSigningProvider& out, std::vector<unsigned char>* cache = nullptr) const override final
         ^                                                                                                                                                                   ~~~~~~~~~
src/script/descriptor.cpp:485:10: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
    bool ExpandFromCache(int pos, const std::vector<unsigned char>& cache, std::vector<CScript>& output_scripts, FlatSigningProvider& out) const override final
         ^                                                                                                                                       ~~~~~~~~~
src/script/descriptor.cpp:491:10: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
    void ExpandPrivate(int pos, const SigningProvider& provider, FlatSigningProvider& out) const override final
         ^                                                                                       ~~~~~~~~~
src/script/descriptor.cpp:517:10: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
    bool IsSolvable() const override final { return false; }
         ^                  ~~~~~~~~~
src/script/descriptor.cpp:542:10: warning: 'override' is redundant since the function is already declared 'final' [modernize-use-override]
    bool IsSolvable() const override final { return false; }
         ^                  ~~~~~~~~~
```

Reply:
> This is final, but not virtual, so override is already implied. I am not sure why the compiler warns here?!

This GCC behavior fixed for 9.2 ([GCC Bugzilla – Bug 78010](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78010)).

Reply:
> However `clang-tidy` with [modernize-use-override](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-override.html) will tell you that it, along with a bunch of the other overrides added here are redundant:

There is an [`AllowOverrideAndFinal`](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-override.html#cmdoption-arg-allowoverrideandfinal) option:
> If set to non-zero, this check will not diagnose `override` as redundant with `final`. This is useful when code will be compiled by a compiler with warning/error checking flags requiring `override` explicitly on overridden members, such as `gcc -Wsuggest-override`/`gcc -Werror=suggest-override`. Default is _0_.

Reply:
Addressed in the latest [push](https://github.com/bitcoin/bitcoin/pull/16710#issuecomment-593608787).